log2/logfmt.go

```diff
@@ -0,0 +1,39 @@
+package log
+
+import (
+	"io"
+
+	"gopkg.in/logfmt.v0"
+)
+
+type logfmtLogger struct {
+	w io.Writer
+}
+
+// NewLogfmtLogger returns a logger that encodes keyvals to the Writer in
+// logfmt format. The passed Writer must be safe for concurrent use by
+// multiple goroutines if the returned Logger will be used concurrently.
+func NewLogfmtLogger(w io.Writer) Logger {
+	return &logfmtLogger{w}
+}
+
+func (l logfmtLogger) Log(keyvals ...interface{}) error {
+	// The Logger interface requires implementations to be safe for concurrent
+	// use by multiple goroutines. For this implementation that means making
+	// only one call to l.w.Write() for each call to Log. We first collect all
+	// of the bytes into b, and then call l.w.Write(b).
+	for i := 1; i < len(keyvals); i += 2 {
```

Comment:
Because keyvals contains alternating keys (in even indices) and values (in odd indices). This loop only needs to operate on the values, so it can start at index 1.